ast.go

```diff
@@ -145,7 +145,12 @@ func (fa *ASTFunc) generateParameters() []ast.Expr {
 		if p.Type.IsSlice && !p.Type.IsReturnArgument {
 			hasDeclaration = true
 
-			fa.addGoToCSliceConversion(p.Name, p.Type)
+			
+			if _, ok := p.Heuristics["pass_c"]; !ok {
```

Comment:
Ok, I'm beginning to understand. I can't reconstruct what I intended with the heuristics map - lets say I wanted to be future proof.
I'm not entirely sure about the default thing, this would mean that we actually handled all array/slice parameters wrong in the first place.